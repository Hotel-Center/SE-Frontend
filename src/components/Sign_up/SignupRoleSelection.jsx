import Radio from "@mui/material/Radio";
import RadioGroup from "@mui/material/RadioGroup";
import Box from "@mui/material/Box";
import FormControlLabel from "@mui/material/FormControlLabel";
import { Typography, Button, Link } from "@mui/material";

function SignupRoleSelection(props) {
  const options = {
    customer: "I am planning to search, view and reserve hotels",
    company: "I am planning to manage and advertise my hotels",
  };
  return (
    <Box
      sx={{
        display: "flex",
        flexDirection: "column",
        gap: 4,
      }}
    >
      <Typography variant="h6">
        How are you going to use Hotel Center?
      </Typography>
      <RadioGroup
        value={props.role}
        onChange={(event) =>
          props.dispatch({
            type: "set_role",
            value: event.target.value,
          })
        }
      >
        <FormControlLabel
          value="customer"
          control={<Radio />}
          label={options["customer"]}
        ></FormControlLabel>
        <FormControlLabel
          value="company"
          control={<Radio />}
          label={options["company"]}
        ></FormControlLabel>
      </RadioGroup>
      <Box
        sx={{
          display: "flex",
          alignItems: "center",
          gap: 3,
          width: "100%",
        }}
      >
        <Button
          sx={{
            height: "40px",
            width: "45%",
            ml: "auto",
          }}
          variant="contained"
          onClick={props.nextStep}
          disabled={props.role === "unknown"}
        >
          Next
        </Button>
      </Box>
      <Link
        to="./login"
        variant="body2"
        sx={{
          color: "black",
          mt: 2,
        }}
      >
        Already have an account? Sign in
      </Link>
    </Box>
  );
}

export default SignupRoleSelection;
