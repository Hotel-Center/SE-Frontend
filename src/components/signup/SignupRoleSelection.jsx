import Radio from "@mui/material/Radio";
import RadioGroup from "@mui/material/RadioGroup";
import Box from "@mui/material/Box";
import FormControlLabel from "@mui/material/FormControlLabel";
import { Typography, Button, Link } from "@mui/material";

function SignupRoleSelection(props) {
  const options = {
    customer: "I am planning to search, view and reserve hotels",
    manager: "I am planning to manage and advertise my hotels",
  };
  return (
    <Box
      sx={{
        height: "100%",
        display: "flex",
        flexDirection: "column",
      }}
    >
      <Box
        sx={{
          display: "flex",
          flexDirection: "column",
          gap: 4,
        }}
      >
        <Typography variant="h6">
          How are you going to use Hotel Center?
        </Typography>
        <RadioGroup
          value={props.role}
          onChange={(event) =>
            props.dispatch({
              type: "set_role",
              value: event.target.value,
            })
          }
        >
          <FormControlLabel
            value="customer"
            control={<Radio />}
            label={options["customer"]}
          ></FormControlLabel>
          <FormControlLabel
            value="manager"
            control={<Radio />}
            label={options["manager"]}
          ></FormControlLabel>
        </RadioGroup>
      </Box>
      <Link
        href="/login"
        variant="body2"
        sx={{
          color: "black",
          mt: "auto",
        }}
      >
        Already have an account? Sign in
      </Link>
      <Box
        sx={{
          display: "flex",
          alignItems: "center",
          gap: 3,
          mt: 2,
          mb: 2,
          width: "100%",
        }}
      >
        <Button
          sx={{
            height: "40px",
            width: "45%",
            ml: "auto",
          }}
          variant="contained"
          onClick={props.nextStep}
          disabled={props.role === "unknown"}
        >
          Next
        </Button>
      </Box>
    </Box>
  );
}

export default SignupRoleSelection;
