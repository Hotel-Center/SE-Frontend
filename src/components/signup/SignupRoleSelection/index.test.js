import renderer from "react-test-renderer";
import SignupRoleSelection from ".";

describe("Signup Role Selection Step", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<SignupRoleSelection />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
