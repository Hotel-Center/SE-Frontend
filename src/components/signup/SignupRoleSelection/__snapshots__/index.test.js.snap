// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Role Selection Step renders correctly 1`] = `
<div
  className="MuiBox-root css-1sm2s1z"
>
  <div
    className="MuiBox-root css-6r2fzw"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      How are you going to use Hotel Center?
    </h6>
    <div
      className="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
      role="radiogroup"
    >
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            name=":r0:"
            onChange={[Function]}
            required={false}
            type="radio"
            value="customer"
          />
          <span
            className="css-hyxlzm"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          I am planning to search, view and reserve hotels
        </span>
      </label>
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={false}
            className="PrivateSwitchBase-input css-1m9pwf3"
            name=":r0:"
            onChange={[Function]}
            required={false}
            type="radio"
            value="manager"
          />
          <span
            className="css-hyxlzm"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
              data-testid="RadioButtonUncheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
              />
            </svg>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
              data-testid="RadioButtonCheckedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
              />
            </svg>
          </span>
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          I am planning to manage and advertise my hotels
        </span>
      </label>
    </div>
  </div>
  <a
    className="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-eb54fy-MuiTypography-root-MuiLink-root"
    href="/login"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    Already have an account? Sign in
  </a>
  <div
    className="MuiBox-root css-t0kr8q"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-kq49rb-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Next
    </button>
  </div>
</div>
`;
